### PR TITLE
Support generating OpenShift manifests

### PR DESCRIPTION
**Pull request by iocanel**
This pull request brings Openshift support to the kubernetes-extension.

The idea is that the user is able to select among:

- kubernetes
- openshift 
- both

using an option like:

`kubernetes.deployment.target=openshift` in the `application.properties`.

The pull request includes integration-test and docs and also a tiny fix to the existing integration tests.

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Looks good 👍!

Assuming integration tests pass, let's merge

File: docs/src/main/asciidoc/kubernetes-resources.adoc [hunk 1]
```diff
@@ -36,7 +36,8 @@ mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
 == Enable Kubernetes support
 
 Quarkus offers the ability to automatically generate Kubernetes resources based on sane defaults and user supplied configuration. The implementation that takes care
-of generating the actual Kubernetes resources is provided by https://github.com/dekorateio/dekorate/[dekorate].
+of generating the actual Kubernetes resources is provided by https://github.com/dekorateio/dekorate/[dekorate]. Currently it supports the genration of resources for
```
Comment (geoand) [inline docs/src/main/asciidoc/kubernetes-resources.adoc]: `genration` -> `genration`

File: docs/src/main/asciidoc/kubernetes-resources.adoc [hunk 2]
```diff
@@ -392,4 +393,68 @@ Below you will find tables describing all available types.
 | registry | String |   |  |
 | auto-push-enabled | boolean |   | false 
 | auto-build-enabled | boolean |   | false 
+|====
+
+=== Openshift support
+
+To enable the genration of Openshift resources, you need to include Openshift in the target platforms:
```
Comment (geoand) [inline docs/src/main/asciidoc/kubernetes-resources.adoc]: `genration` -> `generation`

File: docs/src/main/asciidoc/kubernetes-resources.adoc [hunk 3]
```diff
@@ -392,4 +393,68 @@ Below you will find tables describing all available types.
...
+
+[source]
+----
+kubernetes.deployment.target=openshift
+----
+
+If you need to genrate resources for both platforms (vanilla kubernetes and opesnhift), then you need to include both.
```
Comment (geoand) [inline docs/src/main/asciidoc/kubernetes-resources.adoc]: `genrate` -> `generate`

File: docs/src/main/asciidoc/kubernetes-resources.adoc [hunk 4]
```diff
@@ -392,4 +393,68 @@ Below you will find tables describing all available types.
...
+
+[source]
+----
+kubernetes.deployment.target=kubernetes openshift
```
Comment (geoand) [inline docs/src/main/asciidoc/kubernetes-resources.adoc]: Is this comma whitespace separated or comma separated?

File: extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java [hunk 5]
```diff
@@ -48,6 +49,10 @@
     private static final String PROPERTY_PREFIX = "dekorate.";
     private static final String ALLOWED_GENERATOR = "kubernetes";
 
+    private static final String DEPLOYMENT_TARGET = "kubernetes.deployment.target";
+    private static final String KUBERNETES = "kubernetes";
+    private static final List<String> DEFAULT_TARGETS = Arrays.asList(KUBERNETES);
```
Comment (geoand) [inline extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: This seems to be unused

**Review by geoand — CHANGES_REQUESTED** [review]
Just some minor comments. I think we can get them in once addressed

File: docs/src/main/asciidoc/kubernetes-resources.adoc (hunk 4 above)
Comment (iocanel) [reply docs/src/main/asciidoc/kubernetes-resources.adoc]: whitespace

File: docs/src/main/asciidoc/kubernetes-resources.adoc [hunk 6]
```diff
@@ -392,4 +393,68 @@ Below you will find tables describing all available types.
...
+=== Openshift support
+
+To enable the generation of Openshift resources, you need to include Openshift in the target platforms:
+
+[source]
+----
+kubernetes.deployment.target=openshift
+----
+
+If you need to generate resources for both platforms (vanilla kubernetes and opesnhift), then you need to include both (space separated).
```
Comment (geoand) [inline docs/src/main/asciidoc/kubernetes-resources.adoc]: Sorry, I just saw that Openshift is spelled wrong here.

File: docs/src/main/asciidoc/kubernetes-resources.adoc (hunk 4 above)
Comment (stuartwdouglas) [reply docs/src/main/asciidoc/kubernetes-resources.adoc]: We use comma's everywhere else, it should also use comma's for consistency.

Is there any reason why this is not using the Quarkus configuration model, but is doing its own config?

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/kubernetes-resources.adoc (hunk 4 above)
Comment (iocanel) [reply docs/src/main/asciidoc/kubernetes-resources.adoc]: Switched to `comma separated`.

Regarding the config, this pull request does get the config from the Quarkus `ConfigProvider` and converts it to the `dekorate` format. Does this anser your question @stuartwdouglas or you meant something else?

Comment (geoand) [reply docs/src/main/asciidoc/kubernetes-resources.adoc]: @stuartwdouglas The configuration of this extension uses the pass-through strategy like the Kafka Streams does. 
Originally this extension had the regular Quarkus configuration but it was too limiting so it was switched as part of #4308 (see [this](https://github.com/quarkusio/quarkus/pull/4308#discussion_r330445280) for more details)

**Comment by gsmet** [comment]
Please make the title a bit more descriptive of what this PR is about as it will end up in the announcement email. Thanks!

**Comment by geoand** [comment]
@stuartwdouglas Have your comments been addressed?

**Review by geoand — APPROVED** [review]

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**